app/javascript/pages/DirectoryPage.tsx

```diff
@@ -23,15 +28,129 @@ interface DirectoryProps {
+
+const getListingImageCardStatuses = (listing: RailsRentalListing): StatusBarType[] => {
+  const statuses: StatusBarType[] = []
+
+  const formattedDueDateString = dayjs(listing.Application_Due_Date).format("MMMM DD, YYYY")
+  const lotteryResultsDateString = dayjs(listing.Lottery_Results_Date).format("MMMM DD, YYYY")
+
+  if (new Date(listing.Application_Due_Date) > new Date()) {
+    return [
+      {
+        status: ApplicationStatusType.Open,
+        content: `${t("listings.applicationDeadline")}: ${formattedDueDateString}`,
+      },
+    ]
+  } else {
+    if (!listing.Publish_Lottery_Results) {
+      statuses.push({
+        status: ApplicationStatusType.Closed,
+        content: `${t("listings.applicationDeadline")}: ${formattedDueDateString}`,
+        hideIcon: true,
+      })
+    }
+    statuses.push({
+      status: ApplicationStatusType.PostLottery,
+      content: `${t("listings.lotteryResults.cardTitle")}: ${lotteryResultsDateString}`,
+      hideIcon: true,
+    })
+  }
+
+  return statuses
+}
+
+const getNumberString = (currencyNumber: number) =>
+  currencyNumber ? currencyNumber.toLocaleString() : null
+
+const getRangeString = (min: number, max: number, suffix: string, prefix?: string) => {
+  if (min && max && min !== max) {
+    return `${prefix ?? ""}${getNumberString(min)} to ${prefix}${getNumberString(max)}${suffix}`
+  }
+  if (min || max) {
+    return `${prefix ?? ""}${getNumberString(min ?? max)}${suffix}`
+  }
+  return null
+}
+
+const getRentRangeString = (summary: RailsRentalUnitSummary) => {
+  const rentRangeString = getRangeString(
+    summary.minMonthlyRent,
+    summary.maxMonthlyRent,
+    " per month",
+    "$"
+  )
+  const percentIncomeRangeString = getRangeString(
+    summary.minPercentIncome,
+    summary.maxPercentIncome,
+    "% income"
+  )
+  return rentRangeString ?? percentIncomeRangeString ?? ""
+}
+
+const getUnitSummaryTable = (listing: RailsRentalListing) =>
+  listing.unitSummaries.general
+    .filter((summary) => !!summary.unitType)
+    .map((summary) => ({
+      unitType: <>{summary.unitType}</>,
+      minimumIncome: (
+        <>
+          {getRangeString(summary.absoluteMinIncome, summary.absoluteMaxIncome, " per month", "$")}
+        </>
+      ),
+      rent: <>{getRentRangeString(summary)}</>,
+    }))
+
+const getTableHeader = (listing: RailsRentalListing) => {
+  let header = null
+  if (listing.Units_Available > 0) {
+    header = "Available Units"
+  }
+  if (listing.hasWaitlist) {
+    header = header ? `${header} & Open Waitlist` : "Open Waitlist"
+  }
+  return header
+}
+
+type Listing = RailsRentalListing & {
+  Reserved_community_type: string
 }
 
+const getListings = (listings) =>
+  listings.map((listing: Listing, index) => (
+    <ListingCard
+      key={index}
+      imageCardProps={{
+        imageUrl: listing.imageURL,
+        subtitle: `${listing.Building_Street_Address}, ${listing.Building_City} ${listing.Building_State}, ${listing.Building_Zip_Code}`,
```

Comment:
this was an issue before this PR too, but if any of these are undefined they show up as "undefined" would be nice to just e.g. leave out the zip code if not available